modules/ROOT/pages/worker-monitoring.adoc

```diff
@@ -1,102 +1,93 @@
+image:logo-rtf-disabled.png[link="/runtime-fabric/deploy-to-runtime-fabric", title="Runtime Fabric"]
 
-[NOTE]
-Worker Monitoring is a feature that is only available for applications deployed to CloudHub workers. It is not available for applications that you deploy using the Runtime Manager to local servers. See xref:deployment-strategies.adoc[Deployment Strategies] and xref:monitoring.adoc[Monitoring Applications] for more details.
 
-CloudHub supports worker monitoring and automatic restarts for unresponsive applications. Monitoring allows you the option to receive notifications and diagnostic information if your application becomes unresponsive. Optionally, enable automatic restarts so that your application can recover automatically without your intervention. CloudHub gives you the flexibility to conduct monitoring with or without enabling automatic restarts.
+CloudHub monitors all applications and enables you to receive notifications and diagnostic information if your application becomes unresponsive.
 
-Monitoring is enabled by default on all applications deployed on CloudHub. Applications already deployed before the monitoring feature was introduced need to be redeployed for monitoring to take effect. 
+You can enable automatic restarts so that your application can recover automatically without your intervention.
 
-You can activate the automatic restart feature on a per-application basis using the Runtime Manager console when you deploy a new application or redeploy an existing application.
 
-== Prerequisites
+== How Application Monitoring Works
 
-This document assumes that you are familiar with xref:deploying-to-cloudhub.adoc[Deploying to CloudHub] and that you are deploying to runtime 3.4.n, 3.5.n, or newer.
+Monitoring works differently depending on your Mule version.
 
-== Enabling Automatic Restarts
+* Mule 4.x
++
+Every 30 seconds, CloudHub checks the PID of the Mule process.
++
+If CloudHub detects the application process is not running three times in a row, the logs print a warning.
 
-During deployment, you have the option to enable automatic restarts for your application.
+* Mule 3.8.x
++
+When you deploy, CloudHub inserts a monitoring ping flow into your application.
++
+When the application starts, the logs display a message to indicate that monitoring is enabled and that the ping flow has been injected into your application:
++
+.The arrow shows the `Monitoring enabled` and `Registering ping flow injector` messages in the logs.
+image::cloudhub-app-monitor-enabled.png[Monitoring enabled and Registering ping flow injector message in the logs]
++
+Every 30 seconds, CloudHub sends an HTTP GET request to the ping flow within the application.
++
+If the application is running normally, the ping flow sends a response.
+If CloudHub receives no response from the ping three times in a row, the logs display a warning and a full thread dump.
```

Comment:
```suggestion
If CloudHub receives no response from three consecutive pings, the logs display a warning and a full thread dump.
```

Reply:
Do the logs display the dump or do we mean to say that the dump occurs...or both?